### PR TITLE
feat: Initial project structure

### PR DESCRIPTION
**Pull request by lfvbarcus**
This PR adds initial project structure.

### CODE REVIEW

File: .github/workflows/publish_gh_pages.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+concurrency:
+  group: github-pages
+  cancel-in-progress: false
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+jobs:
+  build:
```
Comment (lfvjimisola) [inline .github/workflows/publish_gh_pages.yml:19]: change job name to `publish-gh-pages`

File: .github/workflows/publish_pypi_prod.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  
+
+  publish-to-pypi:
+    needs: 
+      - check-release
+      - build
+    runs-on: ubuntu-latest
+    environment:
+      name: prod
+      url: https://pypi.org/p/reqstool-client
```
Comment (lfvjimisola) [inline .github/workflows/publish_pypi_prod.yml:23]: wrong url

File: .github/workflows/publish_pypi_test.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    name: Reuse build
+    uses: ./.github/workflows/build.yml
+
+  publish-to-test-pypi:
+    needs: build
+    runs-on: ubuntu-latest
+    # Specifying a GitHub environment is optional, but strongly encouraged
+    environment:
+      name: test
+      url: https://test.pypi.org/p/reqstool-client
```
Comment (lfvjimisola) [inline .github/workflows/publish_pypi_test.yml:17]: wrong url

File: README.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+This documentation provides information on how to use the Reqstool Hatch Plugin. The plugin is designed to be used with the Hatch build tool and facilitates the integration of the Reqstool Decorators in your project.
+
+## Installation
+
+To use the Reqstool Hatch Plugin, follow these steps:
+
+- Update your project dependencies in the `pyproject.toml` file and 
+ensure that the Reqstool Decorators' dependency is listed as follows;
+``` 
+dependencies = ["reqstool-decorators == <version>"]
```
Comment (lfvjimisola) [inline README.md:12]: wrong dependency name

File: docs/antora-playbook.yml [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/asciidoctor/asciidoctor-intellij-plugin/main/src/main/resources/jsonSchemas/antoraPlaybookSchema.json
+
+site:
+  title: Reqstool Python Hatch Plugin Documentation
```
Comment (lfvjimisola) [inline docs/antora-playbook.yml:4]: Reqstool Hatch Plugin Documentation

File: docs/antora.yml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/asciidoctor/asciidoctor-intellij-plugin/main/src/main/resources/jsonSchemas/antoraComponentSchema.json
+
+name: reqstool-python-hatch-plugin
+title: Reqstool Python Hatch Plugin
+version: 0.0.1
```
Comment (lfvjimisola) [inline docs/antora.yml:5]: bump version so that it's unique

Comment (lfvbarcus) [reply docs/antora.yml:5]: fixed

File: docs/antora-playbook.yml (hunk 5 above)
Comment (lfvbarcus) [reply docs/antora-playbook.yml:4]: Fixed

File: README.md (hunk 4 above)
Comment (lfvbarcus) [reply README.md:12]: fixed

File: .github/workflows/publish_pypi_test.yml (hunk 3 above)
Comment (lfvbarcus) [reply .github/workflows/publish_pypi_test.yml:17]: fixed

File: .github/workflows/publish_pypi_prod.yml (hunk 2 above)
Comment (lfvbarcus) [reply .github/workflows/publish_pypi_prod.yml:23]: fixed

File: .github/workflows/publish_gh_pages.yml (hunk 1 above)
Comment (lfvbarcus) [reply .github/workflows/publish_gh_pages.yml:19]: fixed

**Pull request merged**